spyderlib/widgets/file_switcher.py

```diff
@@ -0,0 +1,405 @@
+# -*- coding: utf-8 -*-
+#
+# Augmented file switcher by DM, 2015 (based on original in widgets/editor.py)
+# Licensed under the terms of the MIT License
+# (see spyderlib/__init__.py for details)
+
+from __future__ import print_function
+
+from spyderlib.baseconfig import _
+from spyderlib.qt.QtGui import (QVBoxLayout, QHBoxLayout,
+                                QListWidget, QListWidgetItem,
+                                QDialog, QLineEdit, QKeyEvent,
+                                QStyledItemDelegate, QStyleOptionViewItemV4, 
+                                QApplication, QTextDocument, QStyle,
+                                QAbstractTextDocumentLayout, QToolButton,
+                                QToolTip)
+from spyderlib.qt.QtCore import (Signal, Qt,  QObject,
+                                  QSize, QEvent, QPoint)
+
+import os
+import os.path as osp
+
+# Local imports
+from spyderlib.guiconfig import  new_shortcut
+from spyderlib.py3compat import to_text_string
+from spyderlib.utils.qthelpers import get_std_icon
+
+class HTMLDelegate(QStyledItemDelegate):
+    """ Taken from http://stackoverflow.com/a/5443112/2399799
+    """
+    def paint(self, painter, option, index):
+        options = QStyleOptionViewItemV4(option)
+        self.initStyleOption(options,index)
+
+        style = QApplication.style() if options.widget is None \
+                                        else options.widget.style()
+
+        doc = QTextDocument()
+        doc.setHtml(options.text)
+
+        options.text = ""
+        style.drawControl(QStyle.CE_ItemViewItem, options, painter);
+
+        ctx = QAbstractTextDocumentLayout.PaintContext()
+
+        textRect = style.subElementRect(QStyle.SE_ItemViewItemText, options)
+        painter.save()
+        painter.translate(textRect.topLeft())
+        painter.setClipRect(textRect.translated(-textRect.topLeft()))
+        doc.documentLayout().draw(painter, ctx)
+
+        painter.restore()
+
+    def sizeHint(self, option, index):
+        options = QStyleOptionViewItemV4(option)
+        self.initStyleOption(options,index)
+
+        doc = QTextDocument()
+        doc.setHtml(options.text)
+        doc.setTextWidth(options.rect.width())
+        return QSize(doc.idealWidth(), doc.size().height())
+
+
+
+class UpDownFilter(QObject):
+    up_down = Signal(int)
+    def eventFilter(self, src, e):
+        if isinstance(e,QKeyEvent) and e.type() == QEvent.KeyPress:
+            if e.key() == Qt.Key_Up:
+                self.up_down.emit(-1)
+            elif e.key() == Qt.Key_Down:
+                self.up_down.emit(+1)
+        #Call Base Class Method to Continue Normal Event Processing
+        return super(UpDownFilter,self).eventFilter(src, e)
+
+def shorten_paths(path_list):
+    """
+    Takes a list of paths and tries to "intelligently" shorten them all.  The
```

Comment:
All this function has multiple PEP8 warnings.

Can you got to the preferences ad activate PEP8 warnings? that way you can check what are the problems and fix them right away.

![image](https://cloud.githubusercontent.com/assets/3627835/7337643/7a164d64-ec31-11e4-8921-a4a1ec49d7f9.png)

Please make sure that any code you wrote or modified is PEP8 compliant, do not change any code you did not include or modify.

This is both easier for you and us when going through the code.